File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.go
func isVowel(r byte) bool {
    VOWELS := map[byte]bool{'a': true, 'e': true, 'i': true, 'o': true, 'u': true}
    return VOWELS[r]
}

func maxVowels(s string, k int) int {
    count := 0
    maxCount := 0
    
    for j, _ := range(s) {
        i := j - k + 1
        if i > 0 && isVowel(s[i - 1]) {
            count -= 1
        }
        if isVowel(s[j]) {
            count += 1
        }
        if count > maxCount {
            maxCount = count
        }
        if maxCount == k {
            return k
        }
    }
    return maxCount
}

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        VOWELS = set("aeiou")
        max_vowel_count = 0
        count = 0
        
        for right in range(len(s)):
            left = right - k + 1
            if left > 0:
                count -= 1 if s[left - 1] in VOWELS else 0
            count += 1 if s[right] in VOWELS else 0
            max_vowel_count = max(max_vowel_count, count)
            
            if max_vowel_count == k:
                return k
            
        
        return max_vowel_count
            
        
        
        